### ISSUE TITLE
Residence custom flags

### ISSUE THREAD

**Issue by MrApik** [PluginBugs/Issues-ItemsAdder#2372]
### Terms

- [X] I'm using the very latest version of ItemsAdder and its dependencies.
- [X] I already searched on this [Github page](https://github.com/PluginBugs/Issues-ItemsAdder/issues) to check if the same suggestion was already reported.
- [X] I already searched on the [plugin wiki](https://itemsadder.devs.beer/) to know if the feature is already implemented.
- [X] I already searched on the **Discord** server to know if anyone already has a solution for this or if the feature was already suggested.

### Discord tag (optional)

MrApik#0007

### Describe the solution you'd like

I would like to see Residence compatibility for Vehicles as it is one of the most common plugins for land-protection.

It would be nice to have:

- [ ] Flag for destroying vehicles
- [ ] Flag for mounting vehicles inside the residence
- [ ] Flag for entering vehicles into the Residence

### Is your feature request related to a problem?

I cannot use this plugin now as everyone is able to mount on the vehicles inside the residences and just steal them.

### Describe alternatives you've considered

I tried to make my own integration into Residence, but API for Vehicles is not available and even vanilla like events are problematic as there are no API methods to verify if the entity is vehicle or not.

**Comment by MrApik** [comment]
Hi, do you have any estimate?

**Comment by LoneDev6** [comment]
Not really, wasn't scheduled for a recent update because wasn't requested by many customers.
Do you need it in a rush? I can re-schedule it if you need it quick.
